### DISCUSSION TITLE
Docker Team?

### DISCUSSION THREAD

**Discussion by tisoft** [SAIC-iSmart-API/documentation#2] [Ideas]
Now that docker has stepped back from ending free team accounts, should we create a Docker team to have all images from the several subprojects under one namespace?

What do you think?

**Comment by tisoft** [comment]
I created a docker organization: https://hub.docker.com/u/saicismartapi

Both the Java and the Python MQTT gateways are pushed there. The credentials are stored as organization secrets. They are currently only available to the projects saic-java-client and saic-python-mqtt-gateway. If any other repo needs them in the future, we can easily make them available there, too.

(@tosate I re-added docker hub push to the python build action, hope that was ok. 😄 )

**Reply by Tonno87** [reply]
Is it possible to create tags to the python gateway? At the moment the python client version ist 1.1.6, but the image of the gateway is only latest. So updating on home assistant is impossible at the moment. 

Or can you invite me to the organisation so I can do it myself?

Edit: Username ist Tonno1987

**Reply by tosate** [reply]
There is the python client with tags and releases since it is uploaded to the [Python Package Index](https://pypi.org/) and used by other projects.
For the gateway implementation and it's docker image there is no versioning yet.

If I get you right, your Home Assistant does not recognize that something has changed with the image if each version has the 'latest' tag?
Can Home Assistant manage your docker containers?

So, we would need to introduce versioning for the gateway and somehow pass this version to Docker Hub?

**Reply by Tonno87** [reply]
Yes, home assistant addons works like docker compose. I have to change the image version in github to force an update, otherwise any user need to reinstall the add-on on any changes. Additionaly they need to fill in all options fields again. 
HASS.OS periodically checks the add-on repository for new versions. So if you release a new version like today, anyone has to create a tag on docker hub. After that I can change the version in github successfully.

**Reply by tosate** [reply]
The MQTT gateway docker image has a version tag now.

**Reply by Tonno87** [reply]
Thanks a lot, updates are working now:
![image](https://user-images.githubusercontent.com/59800400/234671350-4dfc4d62-e0ab-4ff7-90c1-81176d80cfe6.png)
👍
